packages/component/src/providers/internal/SendBox/SendBoxComposer.tsx

```diff
@@ -0,0 +1,150 @@
+
+type ErrorMessageStringMap = ReadonlyMap<SendError, string>;
+
+// TODO: [P2] Complete this component.
+const SendBoxComposer = ({ children }: PropsWithChildren<{}>) => {
+  const [connectivityStatus] = useConnectivityStatus();
+  const [error, setError] = useState<SendError | false>(false);
+  const [sendBoxValue] = useSendBoxValue();
+  const apiSubmitSendBox = useSubmitSendBox();
+  const focus = useFocus();
+  const localize = useLocalizer();
+  const scrollToEnd = useScrollToEnd();
+  const styleToEmotionObject = useStyleToEmotionObject();
+  const submitErrorMessageId = useUniqueId('webchat__send-box__error-message-id');
+  const timeoutRef = useRef<readonly [Timeout, Timeout] | undefined>(undefined);
+
+  const errorMessageStringMap = useMemo<ErrorMessageStringMap>(
+    () =>
+      Object.freeze(
+        new Map<SendError, string>()
+          .set('empty', localize('SEND_BOX_IS_EMPTY_TOOLTIP_ALT'))
+          // TODO: [P0] We should add a new string for "Cannot send message while offline."
+          .set('offline', localize('CONNECTIVITY_STATUS_ALT_FATAL'))
+      ),
+    [localize]
+  );
+  const focusRef = useRefFrom(focus);
+  const scrollToEndRef = useRefFrom(scrollToEnd);
+  const setErrorRef = useRef<typeof setError | undefined>(setError);
+  const submitErrorMessageClassName = styleToEmotionObject(SUBMIT_ERROR_MESSAGE_STYLE) + '';
+  const submitErrorMessageIdState = useMemo<readonly [string | undefined]>(
+    () => Object.freeze([error ? submitErrorMessageId : undefined]) as readonly [string | undefined],
+    [error, submitErrorMessageId]
+  );
+
+  setErrorRef.current = setError;
+
+  const submitErrorRef = useRefFrom<'empty' | 'offline' | undefined>(
+    connectivityStatus !== 'connected' ? 'offline' : !sendBoxValue ? 'empty' : undefined
+  );
+
+  const submit = useCallback<ContextType['submit']>(
+    ({ setFocus } = {}) => {
+      (setFocus === 'main' || setFocus === 'sendBox' || setFocus === 'sendBoxWithoutKeyboard') &&
+        focusRef.current?.(setFocus === 'main' || setFocus === 'sendBox' ? setFocus : 'sendBoxWithoutKeyboard');
+
+      const { current: submitError } = submitErrorRef;
+
+      if (submitError) {
+        timeoutRef.current && timeoutRef.current.forEach(clearTimeout);
+
+        setErrorRef.current?.(false);
+
+        timeoutRef.current = Object.freeze([
+          setTimeout(() => setErrorRef.current?.(submitError), TIME_TO_RESET_ERROR_MESSAGE),
+          setTimeout(() => setErrorRef.current?.(false), TIME_TO_QUEUE_ERROR_MESSAGE)
+        ]) as readonly [Timeout, Timeout];
+      } else {
+        scrollToEndRef.current?.();
+        apiSubmitSendBox();
+      }
+    },
+    [apiSubmitSendBox, focusRef, scrollToEndRef, setErrorRef, submitErrorRef, timeoutRef]
+  );
+
+  useEffect(
+    // Prevent `setTimeout()` from firing after unmount.
+    () => () => {
+      setErrorRef.current = undefined;
+    },
+    [setErrorRef]
+  );
```

Comment:
```suggestion
  useEffect(
    // Prevent `setTimeout()` from firing after unmount.
    () => {
      setErrorRef.current = undefined;
    },
    [setErrorRef]
  );
```

Reply:
This is actually intended.

We want to set `setErrorRef` after *unmount*, not *mount*. So the double `() => () =>` will be called during unmount.